src/core/models.py

```diff
@@ -208,9 +208,9 @@ class Account(AbstractBaseUser, PermissionsMixin):
     email = PGCaseInsensitiveEmailField(unique=True, verbose_name=_('Email'))
     username = models.CharField(max_length=254, unique=True, verbose_name=_('Username'))
 
-    first_name = models.CharField(max_length=300, null=True, blank=False, verbose_name=_('First name'))
-    middle_name = models.CharField(max_length=300, null=True, blank=True, verbose_name=_('Middle name'))
-    last_name = models.CharField(max_length=300, null=True, blank=False, verbose_name=_('Last name'))
+    first_name = models.CharField(max_length=300, null=False, default="", blank=False, verbose_name=_('First name'))
+    middle_name = models.CharField(max_length=300, null=False, default="", blank=True, verbose_name=_('Middle name'))
+    last_name = models.CharField(max_length=300, null=False, default="", blank=False, verbose_name=_('Last name'))
```

Comment:
This is a good catch. i agree these fields should all be `null=False` because they are CharField. 

However, I don't think the `default=""` is necessary because Django's default empty storage is an empty string. See the [documentaiton on null](https://docs.djangoproject.com/en/4.2/ref/models/fields/#null). I could be wrong about this but I think that's how it works.

Also, if we are going to allow the `first_name` field to be empty, it needs to have `blank=True` or otherwise people will continue to get form validation errors.

Reply:
We still need the `default` parameter, otherwise I think object creation will still fail when no value is provided and I'd argue we also need to still set  `blank=True` so that the django forms in the admin area don't declare empty values as invalid.

Reply:
As I understand it, `default` and `blank` work hand-in-hand, right? If we set `blank=True`, then Django will not care if nothing is entered in the field, and on save, it will set the field in the database to an empty string. I *believe* this will happen whether or not default is set, so long as `blank` is `True`.

Reply:
> As I understand it, `default` and `blank` work hand-in-hand, right? If we set `blank=True`, then Django will not care if nothing is entered in the field, and on save, it will set the field in the database to an empty string. I _believe_ this will happen whether or not default is set, so long as `blank` is `True`.

Not quite, while `blank` is declared on the model, `ModelForm` and `AdminForm` will use it to determine if a field is required or not during form validation (`form.is_valid`) and before the instance of the model is created (before the `default` value can be used during `form.save()`). So even if there is a default provided, it is best to also be explicit and declare `blank=True` as to allow forms using this model to leave the field blank.

Here is the offending line on `ModelForm`: https://github.com/django/django/blob/07cc014cb30bc3c343a25c81aad6820bbc72c0d9/django/forms/models.py#L355

Reply:
> Not quite: while blank is declared on the model, ModelForm and AdminForm will use it to determine if a field is required or not during form validation (form.is_valid) and before the instance of the model is created (before the default value can be used during form.save()). So even if there is a default provided, it is best to also be explicit and declare blank=True as to allow forms using this model to leave the field blank.

To be clear, you and I are saying the same thing about `blank=True`. We both think we should set this.

I was only saying that when we do that, we might not need `default=""`, but really I don't mind if we keep it.

Reply:
In 10abe982354c11e29ceb656d0963df6e96b4c6a4 I've left only `first_name` with `blank=True` (AFAICT this go towards the comments collected in Discord).

WRT leaving `default=...` etc., I prefer to see them explicit, although redundant when they are the default values, so that I don't forget, but it's not a strong opinion :slightly_smiling_face: